### ISSUE TITLE
Puma write 500 after persistent connection send SYN

### ISSUE THREAD

**Issue by xuxiangyang** [puma/puma#2089]
**Describe the bug**
With http 1.1, puma write 500 after peer persistent tcp send FIN, it should just send FIN back

![image](https://user-images.githubusercontent.com/4926529/70893220-9aba4980-2025-11ea-9b83-7b00bbd65df1.png)

**Puma config:**

just run in development mode

```bash
puma
```

**To Reproduce**
```
run lambda { |env| [200, {"Content-Type" => "text/plain"}, ["Hello World"]] }
```

Run it with:

```
bundle exec puma hello.ru
```

I use [net-http-persistent](https://github.com/drbrain/net-http-persistent) to perform http 1.1 request, wireshark to see the tcp traffic.

```
require 'net/http/persistent'
http = Net::HTTP::Persistent.new
http.request URI("http://127.0.0.1:9292/ping")
http.shutdown
```

**Expected behavior**
I think Puma should just send a FIN back, and client should not send RST

**Desktop (please complete the following information):**
 - OS: Mac 10.15.2
 - Puma Version:  Version 3.12.1 (ruby 2.5.3-p105), codename: Llamas in Pajamas

**Comment by xuxiangyang** [comment]
I think puma sent 500 from this line https://github.com/puma/puma/blob/master/lib/puma/reactor.rb#L228

**Comment by dentarg** [comment]
>this line

Permalink with nice preview:

https://github.com/puma/puma/blob/51494b054797f78ef2e936c449791e2b6ceee647/lib/puma/reactor.rb#L224-L229

**Comment by nateberkopec** [comment]
If someone could link the relevant line of the relevant spec we’re supposed to follow too that would be ❤️

**Comment by praboud** [comment]
FWIW, I can't reproduce this behaviour. I was using Ruby 2.6.3-p62, on Puma v3.12.1 (461c9e99783e5f69e632acedae83be55017d5fe4) on OS X 10.14.6. I followed the same `bundle exec puma hello.ru` + `net-http-persistent` repro procedure described above, but didn't get any weird 500s in Wireshark:

![image](https://user-images.githubusercontent.com/747924/78749899-04879780-7924-11ea-8aa4-6e7ab1f2a895.png)

**Comment by avlazarov** [comment]
Here are the steps I take to get in order to reproduce the issue in 100% of the times:

```
→ puma hello.ru
Puma starting in single mode...
* Version 4.3.5 (ruby 2.7.0-p0), codename: Mysterious Traveller
* Min threads: 0, max threads: 16
* Environment: development
* Listening on tcp://0.0.0.0:9292
Use Ctrl-C to stop
```

```
require 'socket'
socket = TCPSocket.new('0.0.0.0', 9292)
sleep 1
socket.close
```

Same result when using https://github.com/httprb/http (persistent) and ruby 2.7.0
```
require 'http'
connection = HTTP.persistent 'http://localhost:9292', timeout: 50
connection.get('/ok')
sleep 1
connection.close
```

[TCP dump](https://gist.github.com/avlazarov/cde88b0f069089ac5e1320b7ae780c2f)

Ubuntu 20.04

**Comment by nateberkopec** [comment]
Reproing locally, it looks like this issue was fixed by 5.0.3.
